# Data Analytics

Data yang digunakan pada analisis data ini merupakan data hasil panen tanaman padi di Sumatra
Dataset diperoleh pada website "https://www.kaggle.com/datasets/ardikasatria/datasettanamanpadisumatera/data"

## Business Goals
1. Mencari korelasi antara produksi pada setiap provinsi
2. Mencari pengaruh terbesar dari hasil produksi
3. Mencari variabel yang tidak mempengaruhi hasil produksi
4. Mencari tahun produksi terbesar
5. Mencari cara untuk mengembangkan hasil produksi
6. Penggunaan metode regresi linear untuk memprediksi nilai produksi dengan variabel yang mempengaruhinya

### Library Import

In [52]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np

%matplotlib inline

### Load Dataset

In [53]:
file_path = "datasets/data_panen_padi.csv"
df = pd.read_csv(file_path)
print("Full train dataset shape is {}".format(df.shape))

Full train dataset shape is (224, 7)


Data yang diteliti terdiri dari 224 baris dan 7 kolom. Selanjutnya, kita bisa melihat tabel data kita dengan memanggil 3 baris pertama


## Understanding Data

In [54]:
df.head(3)

,Provinsi,Tahun,Produksi,Luas Panen,Curah hujan,Kelembapan,Suhu rata-rata
0,Aceh,1993,1329536.0,323589.0,1627.0,82.00,26.06
1,Aceh,1994,1299699.0,329041.0,1521.0,82.12,26.92
2,Aceh,1995,1382905.0,339253.0,1476.0,82.72,26.27


Data tersebut terdiri dari 1 kategori, 1 time-series, dan 5 data numerikal. Fitur pada data ini bisa kita gunakan untuk mengalisis indikasi yang berhubungan dengan kolom target kita yaitu "produksi".

In [55]:
df.info()
df.describe()
df.nunique()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 224 entries, 0 to 223
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Provinsi        224 non-null    object 
 1   Tahun           224 non-null    int64  
 2   Produksi        224 non-null    float64
 3   Luas Panen      224 non-null    float64
 4   Curah hujan     224 non-null    float64
 5   Kelembapan      224 non-null    float64
 6   Suhu rata-rata  224 non-null    float64
dtypes: float64(5), int64(1), object(1)
memory usage: 12.4+ KB


Provinsi            8
Tahun              28
Produksi          224
Luas Panen        224
Curah hujan       220
Kelembapan        180
Suhu rata-rata    136
dtype: int64

Data tersebut tidak memiliki nilai kosong pada setiap barisnya.

## Data Cleaning

Sebelum melakukan handle values, missing, dan sebagainya. Namun usahakan untuk membuat data dummy agar data orisinalmya tidak hilang.

In [56]:
data = df.copy()
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 224 entries, 0 to 223
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Provinsi        224 non-null    object 
 1   Tahun           224 non-null    int64  
 2   Produksi        224 non-null    float64
 3   Luas Panen      224 non-null    float64
 4   Curah hujan     224 non-null    float64
 5   Kelembapan      224 non-null    float64
 6   Suhu rata-rata  224 non-null    float64
dtypes: float64(5), int64(1), object(1)
memory usage: 12.4+ KB


### Column Lowerspace

In [57]:
data.columns = map(str.lower, data.columns)
data.head(3)

,provinsi,tahun,produksi,luas panen,curah hujan,kelembapan,suhu rata-rata
0,Aceh,1993,1329536.0,323589.0,1627.0,82.00,26.06
1,Aceh,1994,1299699.0,329041.0,1521.0,82.12,26.92
2,Aceh,1995,1382905.0,339253.0,1476.0,82.72,26.27


Judul pada setiap kolom dibuat lowercase dengan tujuan untuk memastikan konsistensi, keterbacaan, dan mencegah adanya error. Selanjutnya kitaTidak ada nilai kosong sama sekali pada data, selanjutnya kita ubah nama setiap kolom menjadi lowercase


### Handling Missing

In [58]:
data.isna().sum()

provinsi          0
tahun             0
produksi          0
luas panen        0
curah hujan       0
kelembapan        0
suhu rata-rata    0
dtype: int64

Tidak terdapat nilai kosong

### Handling Duplicate

In [59]:
data = data.drop_duplicates(keep='first')
data.shape

(224, 7)

## Exploratory Data Analysis